File: oop/heritage/Tp3/src/Professor.java
public class Professor extends Employe {
    String speciality;

    public Professor(String lastname, String firstname, double salary, String speciality) {
        super(lastname, firstname, salary);
        this.speciality = speciality;
    }

    public String getspeciality() {
        return speciality;
    }

    public void setspeciality(String speciality) {
        this.speciality = speciality;
    }

    @Override
    public String toString() {
        return "Professor [speciality=" + speciality + ", salary=" + salary + ", Id=" + getId() + ", lastname=" + getLastname()
                + ", firstname=" + getFirstname() + "]";
    }
    
}

File: oop/heritage/CPP/exercice2/src/Professor.cpp
#include "Professor.hpp"

Professor::Professor(const std::string& lastname, const std::string& firstname, double salary, const std::string& speciality)
    : Employe(lastname, firstname, salary), speciality(speciality) {}

const std::string& Professor::getSpeciality() const {
    return speciality;
}

void Professor::setSpeciality(const std::string& newSpeciality) {
    speciality = newSpeciality;
}

std::string Professor::toString() const {
    return Employe::toString() + "Professor [speciality=" + speciality + "]";
}
